edwb_app_intro.py

```diff
@@ -171,78 +172,46 @@ def get_related_slides(course_name, slide_name, lno):
-
+ 
 
-        if next_slide_name is not None:
-            set_sess(request.url,ses_disp_str)
-
-        return render_template("slide.html",slide_name=next_slide_name,course_name=course_name,num_related_slides=num_related_slides,related_slides = related_slides,disp_str=disp_str,disp_color=disp_color,disp_snippet=disp_snippet,related_course_names=related_course_names,lno=lno,lec_name=lec_name,lec_names=lec_names,lnos=lnos,course_names=COURSE_NAMES,num_courses=NUM_COURSES,rel_lnos=rel_lnos,rel_lec_names=rel_lec_names,vis_urls=vis_urls,vis_strs=vis_strs,num_vis=NUM_VIS,video_link=video_link,lec_slides=lec_slides,base_url = config.base_url, pdf_url= config.pdf_url, textbook_link= textbook_link)
-    except:
-        return render_template("notFound.html")
-
-
-@app.route('/related_slide/<course_name>/<lno>/<slide_name>')
-def related_slide(course_name,slide_name,lno):
+@app.route('/get_search_slide/<course_name>/<lno>/<slide_name>/<idx>')
+def get_search_slide(course_name,slide_name,lno, idx):
     global NUM_VIS
     next_slide_name,lno,lec_name,(num_related_slides,related_slides,disp_str,related_course_names,rel_lnos,rel_lec_names,disp_color,disp_snippet),lec_names,lnos,ses_disp_str,video_link, lec_slides, textbook_link =resolve_slide(course_name,lno,'related',slide_name=slide_name)
     vis_urls,vis_strs = get_prev_urls()
-
-
+    # 1-indexed idx
+    log_helper('get_searched_slide_' + str(idx), "//get_search_slide/", slide_name)
     if next_slide_name is not None:
         set_sess(request.url,ses_disp_str)
 
     return render_template("slide.html",slide_name=next_slide_name,course_name=course_name,num_related_slides=num_related_slides,related_slides = related_slides,disp_str=disp_str,disp_color=disp_color,disp_snippet=disp_snippet,related_course_names=related_course_names,lno=lno,lec_name=lec_name,lec_names=lec_names,lnos=lnos,course_names=COURSE_NAMES,num_courses=NUM_COURSES,rel_lnos=rel_lnos,rel_lec_names=rel_lec_names,vis_urls=vis_urls,vis_strs=vis_strs,num_vis=NUM_VIS,video_link=video_link,lec_slides=lec_slides,base_url = config.base_url, pdf_url= config.pdf_url, textbook_link=textbook_link)
 
 
-
-@app.route('/next_slide/<course_name>/<lno>/<curr_slide>')
-def next_slide(course_name,lno,curr_slide):
+@app.route('/related_slide/<course_name>/<lno>/<slide_name>/<idx>')
+def related_slide(course_name,slide_name,lno, idx):
     global NUM_VIS
-    next_slide_name,lno,lec_name,(num_related_slides,related_slides,disp_str,related_course_names,rel_lnos,rel_lec_names,disp_color,disp_snippet),lec_names,lnos,ses_disp_str,video_link, lec_slides = resolve_slide(course_name,lno,'next',slide_name=curr_slide)
-
+    next_slide_name,lno,lec_name,(num_related_slides,related_slides,disp_str,related_course_names,rel_lnos,rel_lec_names,disp_color,disp_snippet),lec_names,lnos,ses_disp_str,video_link, lec_slides, textbook_link =resolve_slide(course_name,lno,'related',slide_name=slide_name)
     vis_urls,vis_strs = get_prev_urls()
-
-
+    #1 indexed idx
```

Comment:
What does this comment mean?

Reply:
I added a more detailed comment and it should have been zero indexed